kernel/tlb_shootdown/src/lib.rs

```diff
@@ -85,8 +82,15 @@ pub fn send_tlb_shootdown_ipi(my_lapic: &mut LocalApic, pages_to_invalidate: Pag
     *TLB_SHOOTDOWN_IPI_PAGES.write() = Some(pages_to_invalidate);
     TLB_SHOOTDOWN_IPI_COUNT.store(cpu_count - 1, Ordering::SeqCst); // -1 to exclude this core 
 
-    // let's try to use NMI instead, since it will interrupt everyone forcibly and result in the fastest handling
-    my_lapic.send_nmi_ipi(LapicIpiDestination::AllButMe); // send IPI to all other cores but this one
+    #[cfg(target_arch = "x86_64")] {
+        let my_lapic = apic::get_my_apic().expect("If there are more than one CPU ready, this one should be registered");
```

Comment:
a bit clearer, plus "BUG"
```suggestion
        let my_lapic = apic::get_my_apic()
            .expect("BUG: broadcast_tlb_shootdown(): couldn't get LocalApic");
```